internal/folders/manager.go

```diff
@@ -0,0 +1,179 @@
+package folders
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"github.com/go-chi/chi/v5"
+	"github.com/sheltertechsf/sheltertech-go/internal/db"
+)
+
+type Manager struct {
+	DbClient *db.Manager
+}
+
+func New(dbManager *db.Manager) *Manager {
+	manager := &Manager{
+		DbClient: dbManager,
+	}
+	return manager
+}
+
+// Get lists folders for current user 
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Get Folders for current User
+//	@Description	get folders for user
+//	@Tags			folders
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{array}	folders.Folders
+//	@Router			/folders [get]
+func (m *Manager) Get(w http.ResponseWriter, r *http.Request) {
+	userId, err := strconv.Atoi(chi.URLParam(r, "user_id"))
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+
+	dbFolders := m.DbClient.GetFolders(userId)
+	response := Folders{
+		Folders: FromDBTypeArray(dbFolders),
+	}
+	writeJson(w, response)
+}
+
+// Create folder for current user 
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Create Folder for current User
+//	@Description	new folder for user
+//	@Tags			folders
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{object}	folders.Folder
+//	@Router			/folders [post]
+func (m *Manager) Post(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	body, _ := ioutil.ReadAll(r.Body)
+
+	folder := &Folder{}
+	err := json.Unmarshal(body, folder)
+	if err != nil {
+		writeStatus(w, http.StatusInternalServerError)
+	}
+
+	dBFolder := &db.Folder{
+		Name:   folder.Name,
+		Order:  folder.Order,
+		UserId: folder.UserId,
+	}
+
+	err = m.DbClient.CreateFolder(dBFolder)
+	if err != nil {
+		log.Print(err)
+		writeStatus(w, http.StatusInternalServerError)
+	}
+
+	writeStatus(w, http.StatusCreated)
+}
+
+// Get folder by ID
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Get folder by ID
+//	@Description	get current folder for user
+//	@Tags			folders
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{object}	folders.Folder
+//	@Router			/folders/{id} [get]
+func (m *Manager) GetByID(w http.ResponseWriter, r *http.Request) {
+	folderId, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		log.Printf("%v", err)
+	}
+	dbFolder := m.DbClient.GetFolderById(folderId)
```

Comment:
This needs some error handling. I did a local test where I intentionally tried to get a folder with a nonexistent ID, and it causes a nil pointer dereference error. We should gracefully 404 in this case.

Reply:
I fixed this in https://github.com/ShelterTechSF/sheltertech-go/pull/115/commits/6b9dfea690bb046a4c376b06e1e7897d619add71